src/main/java/com/example/shopproject/controller/AdminController.java

```diff
@@ -0,0 +1,76 @@
+package com.example.shopproject.controller;
+
+import com.example.shopproject.entity.Product;
+import com.example.shopproject.entity.Product_category;
+import com.example.shopproject.service.ProductService;
+import com.example.shopproject.service.Product_categoryService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import javax.validation.Valid;
+import java.util.List;
+
+@RestController
```

Comment:
according to rest routing convention you should add /api/v1 before every API, versioning will help you to make new versions of your controllers

Reply:
please create a new separate controller for product and for product categories